### PR TITLE
Enable DirectWeightedAttribute optmization also for arrays.

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst or @toregge PR

### CODE REVIEW

**Review by toregge — APPROVED** [review]

File: searchlib/src/tests/attribute/document_weight_iterator/document_weight_iterator_test.cpp [hunk 1]
```diff
@@ -56,47 +62,57 @@ void populate_string(AttributeVector::SP attr_ptr) {
...
 }
 
+template<CollectionType::Type CT>
 struct LongFixture {
     AttributeVector::SP attr;
     const IDocumentWeightAttribute *api;
-    LongFixture() : attr(make_attribute(BasicType::INT64, CollectionType::WSET, true)),
-                    api(attr->asDocumentWeightAttribute())
+    LongFixture()
+        : attr(make_attribute(BasicType::INT64, CollectionType::WSET, true)),
```
Comment (geirst) [inline searchlib/src/tests/attribute/document_weight_iterator/document_weight_iterator_test.cpp:70]: Shouldn't we use CT instead of CollectionType::WSET?

**Review by geirst — COMMENTED** [review]
👍  But one comment to look at.

File: searchlib/src/tests/attribute/document_weight_iterator/document_weight_iterator_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/tests/attribute/document_weight_iterator/document_weight_iterator_test.cpp:70]: Yes

Comment (baldersheim) [reply searchlib/src/tests/attribute/document_weight_iterator/document_weight_iterator_test.cpp:70]: Fixed in #27653

**Pull request merged**